edgelet/iotedge/src/main.rs

```diff
@@ -55,455 +54,480 @@ async fn run() -> anyhow::Result<()> {
         chrono::Utc::now().format("_%Y_%m_%d_%H_%M_%S_%Z")
     );
 
-    let matches = App::new(crate_name!())
-        .version(&*edgelet_core::version_with_source_version())
+    let matches = Command::new(crate_name!())
+        .version(edgelet_core::version_with_source_version())
         .about(crate_description!())
-        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .subcommand_required(true)
         .arg(
-            Arg::with_name("host")
+            Arg::new("host")
                 .help("Daemon socket to connect to")
-                .short("H")
+                .short('H')
                 .long("host")
-                .takes_value(true)
+                .num_args(1)
                 .value_name("HOST")
                 .global(true)
                 .env("IOTEDGE_HOST")
                 .default_value(default_mgmt_uri),
         )
         .subcommand(
-            SubCommand::with_name("check")
+            Command::new("check")
                 .about("Check for common config and deployment issues")
                 .arg(
-                    Arg::with_name("container-engine-config-file")
+                    Arg::new("container-engine-config-file")
                         .long("container-engine-config-file")
                         .value_name("FILE")
                         .help("Sets the path of the container engine configuration file")
-                        .takes_value(true)
+                        .num_args(1)
+                        .value_parser(clap::value_parser!(OsString))
```

Comment:
`s/OsString/PathBuf/` ? Same for the other `OsString`s that are file paths.

Reply:
bcd8504b8d8ee7b9f0fe6f7402fc5614952b35dc